decko/rails/controllers/card_controller.rb

```diff
@@ -121,13 +121,20 @@ def render_page format, view
     end
   end
 
-  unless Rails.env.development?
-    rescue_from StandardError do |exception|
-      @card ||= Card.new
-      Card::Error.current = exception
-      error = Card::Error.cardify_exception exception, card
-      error.report
-      show error.class.view, error.class.status_code
+  def handle_exception exception
+    @card ||= Card.new
+    Card::Error.current = exception
+    error = Card::Error.cardify_exception exception, card
+    error.report
+    show error.class.view, error.class.status_code
+  end
+
+  class << self
+    def rescue_from_class klass
+      rescue_from(klass) { |exception| handle_exception exception }
     end
   end
+
+  rescue_from_class ActiveRecord::RecordInvalid
+  rescue_from_class(Rails.env.development? ? Card::Error::UserError : StandardError)
```

Comment:
What is the advantage of rescuing UserErrors?

Reply:
This restores the old developer error UX.  If you don't rescue code errors, you get the full backtrace/console thing.  (But you don't really want those for user errors).

Reply:
I think I got the logic wrong when I read it the first time. Makes sense now.